### ISSUE TITLE
Docs give contradictory information about locking in Terraform Enterprise

### ISSUE THREAD

**Issue by BRMatt** [hashicorp/terraform#18146]
On the [remote state](https://www.terraform.io/docs/state/remote.html) part of the documentation it says:

> Terraform Enterprise by HashiCorp is a commercial offering that in addition to locking supports remote operations that allow you to safely queue Terraform operations in a central location. This enables teams to safely modify infrastructure concurrently.

But the [terraform enterprise backend](https://www.terraform.io/docs/backends/types/terraform-enterprise.html) part of the docs explicitly says:

> Kind: Standard (with no locking)

It seems like one of these pages needs to be updated to be in sync with the other?

**Comment by apparentlymart** [comment]
Hi @BRMatt! Sorry for the confusing information here.

The reason for this contradiction is that these two statements are unfortunately using "locking" to mean two different things. :confounded: 

In the world of Terraform CLI backends, "locking" refers to _state locking_, which locks the state file itself (using some means specific to the backend in question) to guard against concurrent operations against that state file.

In the world of Terraform Enterprise, "locking" is a grander concept that applies to a workspace as a whole, ensuring that Terraform Enterprise _itself_ does not attempt to handle two "runs" (a plan followed by an apply) at the same time.

I agree that this is confusing, and is unfortunately a result of Terraform CLI growing its locking feature only after Terraform Enterprise already had its own concept. Thanks for pointing this out; we should indeed revise these docs to properly describe the more advanced idea of "locking" that Terraform Enterprise uses and how it relates to Terraform CLI's state locking feature.

**Comment by mildwonkey** [comment]
I am going to close this issue due to inactivity. The documentation has been updated quite a bit since this issue was initially opened.

If there is still a question, I recommend the [the community forum](https://discuss.hashicorp.com/c/terraform-core), where there are far more people available to help. If there is a bug or you would like to make a feature request, please open a new issue and fill out the template.
Thanks!
